### ISSUE TITLE
Cannot find module 'node:dns'

### ISSUE THREAD

**Issue by xqin** [grpc/grpc-node#2798]
### Problem description
@grpc/grpc-js@1.11.0 breaking support for node v14.

<img width="1101" alt="image" src="https://github.com/user-attachments/assets/65d5aea6-1a10-4a4b-a4ee-c75dd0f3f9e6">

```
➜ node
Welcome to Node.js v14.16.0.
Type ".help" for more information.
> require('@grpc/grpc-js')
Uncaught Error: Cannot find module 'node:dns'
Require stack:
- /Users/test/Desktop/Work/rpc/node_modules/@grpc/grpc-js/build/src/resolver-dns.js
- /Users/test/Desktop/Work/rpc/node_modules/@grpc/grpc-js/build/src/http_proxy.js
- /Users/test/Desktop/Work/rpc/node_modules/@grpc/grpc-js/build/src/transport.js
- /Users/test/Desktop/Work/rpc/node_modules/@grpc/grpc-js/build/src/subchannel-pool.js
- /Users/test/Desktop/Work/rpc/node_modules/@grpc/grpc-js/build/src/internal-channel.js
- /Users/test/Desktop/Work/rpc/node_modules/@grpc/grpc-js/build/src/channel.js
- /Users/test/Desktop/Work/rpc/node_modules/@grpc/grpc-js/build/src/index.js
- <repl>
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/test/Desktop/Work/rpc/node_modules/@grpc/grpc-js/build/src/resolver-dns.js',
    '/Users/test/Desktop/Work/rpc/node_modules/@grpc/grpc-js/build/src/http_proxy.js',
    '/Users/test/Desktop/Work/rpc/node_modules/@grpc/grpc-js/build/src/transport.js',
    '/Users/test/Desktop/Work/rpc/node_modules/@grpc/grpc-js/build/src/subchannel-pool.js',
    '/Users/test/Desktop/Work/rpc/node_modules/@grpc/grpc-js/build/src/internal-channel.js',
    '/Users/test/Desktop/Work/rpc/node_modules/@grpc/grpc-js/build/src/channel.js',
    '/Users/test/Desktop/Work/rpc/node_modules/@grpc/grpc-js/build/src/index.js',
    '<repl>'
  ]
}
```

### Reproduction steps

```

mkdir a
cd a
nvm use 14.16.0
npm init -y
npm i @grpc/grpc-js

node -e 'require("@grpc/grpc-js")'

```

### Environment
 - OS name, version and architecture: macOS
 - Node version v14.16.0
 - Node installation method nvm
 - Package name and version @grpc/grpc-js@1.11.0

**Comment by smalldice** [comment]
same problem!
Why taking this BREAKING CHANGE into a MINOR version???

**Comment by 1anZhang** [comment]
same problem!!!!!!!1

**Comment by murgatroid99** [comment]
This is fixed in version 1.11.1

**Comment by 1412335** [comment]
I faced the same issue when I upgraded to v1.12.0

**Comment by murgatroid99** [comment]
That has been fixed (again) in version 1.12.1
